R/CapStr.R

```diff
@@ -1,6 +1,13 @@
-##### Capitalise input string
+##### Capitalize input string
+#' Capitalize input string
+#'
+#' @param y Input
+#'
+#' @return Capitalized string.
+#' @export
+#'
 CapStr <- function(y) {
-  c <- strsplit(y, " ")[[1]]
-  paste(toupper(substring(c, 1,1)), substring(c, 2),
+  c <- Biostrings::strsplit(y, " ")[[1]]
```

Comment:
> but they generally work the same.

Still worth updating these to come via `base`?

> I'll look into replacing those rapportools calls with something else.

In case this helps - I had taken a note of where rapportools is being used as we had discussed to remove this dependency.

```
civicDrugTable.R 
densityPlot.R
exprGroupsStats_geneWise.R
cumsum_ordered.R
```

Reply:
Yes, definitely, `base` is cleaner!
And yep, noticed that too, thanks!

Reply:
@pdiakumis - just trying to remember where we are with this? Think once we have moved away from rapportools, then this is ready to be merged? Do you want me to have a look if you are under the pump with other work? 

Also, in terms of next steps, should this be a priority https://github.com/umccr/RNAsum/pull/60#issuecomment-1135425644 or you would suggest anoy other outstanding issues?

Losing a bit of momentum here between PRs and discussions, so wanted to make sure it doesn't slip my mind..

Reply:
Ah, apologies, didn't know you were waiting on me and didn't want to mess up this PR code with my changes.
Give me until tomorrow and I'll push some changes to this PR.
I'll start work on the conda recipes this week, good idea.

Reply:
Ok, copied that along with a couple other functions it depends on into `R/utils.R`.
I cleaned up quite a few function calls and definitions that were either referring to variables defined in the report, or where variables were being declared without getting used.
Also sorted out all those global variables - trick in general is to use `.data$foo` when trying to do something on the `foo` column via a pipeline.

Reply:
There are still a couple notes in the check - one about `knitr:::hook_plot_md_base`, and the big one about the `non-portable file paths`. I'll sort out the former at least, and need to play with the data a bit to figure out a better way to organise.

The package size is also pretty big due to the `inst/rawdata` (~750MB), so we'll need to do something about that (e.g. DVC). But that can wait for later.

Reply:
Thanks @pdiakumis - these are really helpful. 
I'll look into resolving `the big one about the non-portable file paths.`.

Reply:
@pdiakumis - changes look good to me. Thanks.

If you're okay approving this PR, I'll merge it and open a new one to resolve for data path issue.

Reply:
Yep, approved thanks, that makes sense!